Description: extern instead ENGINE_API because now static linked
.
Added support for the content of the demo version of the game.
Forwarded: not-needed
--- a/SamTFE/Sources/SeriousSam/SeriousSam.cpp	2024-02-14 22:53:00.349090113 +0300
+++ b/SamTFE/Sources/SeriousSam/SeriousSam.cpp	2024-02-14 23:08:49.377669933 +0300
@@ -63,10 +63,11 @@
 extern FLOAT _fGlobalButtonAdjuster;
 extern FLOAT _fGlobalProfileFOVAdjuster;
 #ifdef PLATFORM_UNIX
-ENGINE_API FLOAT _fWeaponFOVAdjuster;
-ENGINE_API FLOAT _fPlayerFOVAdjuster;
-ENGINE_API FLOAT _fArmorHeightAdjuster;
-ENGINE_API FLOAT _fFragScorerHeightAdjuster;
+extern FLOAT _fWeaponFOVAdjuster;
+extern FLOAT _fPlayerFOVAdjuster;
+extern FLOAT _fArmorHeightAdjuster;
+extern FLOAT _fFragScorerHeightAdjuster;
+extern INDEX sys_iGameDemoMode;
 #else
 extern ENGINE_API FLOAT _fPlayerFOVAdjuster;
 extern ENGINE_API FLOAT _fWeaponFOVAdjuster;
@@ -110,7 +110,7 @@
 __extern INDEX sam_iDisplayDepth  = 0;  // 0==default, 1==16bit, 2==32bit
 __extern INDEX sam_iDisplayAdapter = 0;
 #ifdef SE1_VULKAN
-__extern INDEX sam_iGfxAPI = 1; // default opengl //1;         // 1==Vulkan
+__extern INDEX sam_iGfxAPI = 0; // default opengl //1;         // 1==Vulkan
 #else
 __extern INDEX sam_iGfxAPI = 0;                                // 0==OpenGL
 #endif // SE1_VULKAN
@@ -735,6 +735,17 @@
     FatalError("%s", strError);
   }
 
+  // Check Demo version mode
+  #ifdef PLATFORM_UNIX
+  if(sys_iGameDemoMode == 1) {
+    #ifdef FIRST_ENCOUNTER  // First Encounter
+      sam_strFirstLevel = "Levels\\KarnakDemo.wld";
+    #else
+      sam_strFirstLevel = "Levels\\LevelsMP\\Demo\\Palenque_Demo.wld";
+    #endif
+  }
+  #endif
+
   // always disable all warnings when in serious sam
   _pShell->Execute( "con_bNoWarnings=1;");
 
@@ -1586,7 +1586,7 @@
 #else
     CTString strCommand;
     if (sys_iSysPath == 1) {
-      strCommand = sam_strGameName;
+      strCommand = sam_strGameName + "-vk";
     } else {
       strCommand = "SeriousSam";
     }
--- a/SamTSE/Sources/SeriousSam/SeriousSam.cpp	2024-02-14 22:53:00.349090113 +0300
+++ b/SamTSE/Sources/SeriousSam/SeriousSam.cpp	2024-02-14 23:08:49.377669933 +0300
@@ -63,10 +63,11 @@
 extern FLOAT _fGlobalButtonAdjuster;
 extern FLOAT _fGlobalProfileFOVAdjuster;
 #ifdef PLATFORM_UNIX
-ENGINE_API FLOAT _fWeaponFOVAdjuster;
-ENGINE_API FLOAT _fPlayerFOVAdjuster;
-ENGINE_API FLOAT _fArmorHeightAdjuster;
-ENGINE_API FLOAT _fFragScorerHeightAdjuster;
+extern FLOAT _fWeaponFOVAdjuster;
+extern FLOAT _fPlayerFOVAdjuster;
+extern FLOAT _fArmorHeightAdjuster;
+extern FLOAT _fFragScorerHeightAdjuster;
+extern INDEX sys_iGameDemoMode;
 #else
 extern ENGINE_API FLOAT _fPlayerFOVAdjuster;
 extern ENGINE_API FLOAT _fWeaponFOVAdjuster;
@@ -110,7 +110,7 @@
 __extern INDEX sam_iDisplayDepth  = 0;  // 0==default, 1==16bit, 2==32bit
 __extern INDEX sam_iDisplayAdapter = 0;
 #ifdef SE1_VULKAN
-__extern INDEX sam_iGfxAPI = 1; // default opengl //1;         // 1==Vulkan
+__extern INDEX sam_iGfxAPI = 0; // default opengl //1;         // 1==Vulkan
 #else
 __extern INDEX sam_iGfxAPI = 0;                                // 0==OpenGL
 #endif // SE1_VULKAN
@@ -735,6 +735,17 @@
     FatalError("%s", strError);
   }
 
+  // Check Demo version mode
+  #ifdef PLATFORM_UNIX
+  if(sys_iGameDemoMode == 1) {
+    #ifdef FIRST_ENCOUNTER  // First Encounter
+      sam_strFirstLevel = "Levels\\KarnakDemo.wld";
+    #else
+      sam_strFirstLevel = "Levels\\LevelsMP\\Demo\\Palenque_Demo.wld";
+    #endif
+  }
+  #endif
+
   // always disable all warnings when in serious sam
   _pShell->Execute( "con_bNoWarnings=1;");
 
@@ -1586,7 +1586,7 @@
 #else
     CTString strCommand;
     if (sys_iSysPath == 1) {
-      strCommand = sam_strGameName;
+      strCommand = sam_strGameName + "-vk";
     } else {
       strCommand = "SeriousSam";
     }
